R/data_modify.R

```diff
@@ -262,8 +297,55 @@ data_modify.grouped_df <- function(data, ...) {
     data[rows, ] <- data_modify.data.frame(data[rows, ], ...)
   }
 
+  # check if we have at/modify ----
+  data <- .modify_at(data, .at, .if, .modify, column_names)
+
   # set back attributes and class
   data <- .replace_attrs(data, attr_data)
   class(data) <- class_attr
   data
 }
+
+
+# helper -------------
+
+.modify_at <- function(data, .at, .if, .modify, column_names) {
+  # make sure either .at or .if is defined, not both
+  if (!is.null(.at) && !is.null(.if)) {
+    insight::format_error("You cannot use both `.at` and `.if` at the same time.")
+  }
+  if ((!is.null(.at) || !is.null(.if)) && !is.null(.modify)) {
+    # if we have ".if" defined, specify ".at"
+    if (!is.null(.if)) {
+      .at <- column_names[vapply(data[column_names], .if, logical(1))]
+    }
+    # make sure "modify" is a function
+    if (!is.function(.modify)) {
+      insight::format_error("`.modify` must be a function.")
+    }
```

Comment:
You can move this check earlier, before the first `if` condition so that you don't have to check `!is.null(.modify)`